src/image/prepare_single_image_build_matrix.py

```diff
@@ -57,16 +63,37 @@ def validate_image_trigger(data: dict) -> None:
     for img_number, _ in enumerate(builds):
         builds[img_number]["name"] = args.oci_path.rstrip("/").split("/")[-1]
         builds[img_number]["path"] = args.oci_path
-        # make sure every build of this image has a unique identifier
-        # within the execution of the workflow - use revision number
         builds[img_number]["revision"] = img_number + int(args.next_revision)
 
+        if args.infer_image_track:
+            import sys
+            sys.path.append("src/")
+            from git import Repo
+            from tempfile import TemporaryDirectory as tempdir
+            from uploads.infer_image_track import get_base_and_track
+            with tempdir() as d:
+                url = f"https://github.com/{builds[img_number]['source']}.git"
+                repo = Repo.clone_from(url, d)
```

Comment:
add arg `depth=1` to clone_from function to ensure it's a shallow clone.

Reply:
This shouldn't be a shallow clone, at least not recognised by the original implementation.
https://github.com/canonical/oci-factory/blob/8b5294d8bf3e22c1998928df5f8140a0f132d072/.github/workflows/Image.yaml#L200